pymatgen/io/lammps/inputs.py

```diff
@@ -120,31 +220,66 @@ def as_dict(self):
         d["@class"] = type(self).__name__
         return d
 
+    def __str__(self):
+        return self.get_string()
+
     @staticmethod
-    def from_file(path: Union[str, Path]):
+    def _clean_lines(string_list: list):
         """
-        Creates an InputFile object from a file.
+        Helper method to strips whitespaces, carriage returns and empty
+        lines from a list of strings.
 
         Args:
-            path: Filename to read, including path.
+            string_list (list): List of strings.
+        """
+        for s in string_list:
+            clean_s = s
+            clean_s = clean_s.strip()
+            if not (clean_s == "" or clean_s == "\n"):
+                yield clean_s
 
-        Returns:
-            InputFile
+    def _add_stage_name(self, stage_name: str = None) -> None:
         """
-        filename = path if isinstance(path, Path) else Path(path)
-        with zopen(filename, "rt") as f:
-            return LammpsInputFile.from_string(f.read())
+        Helper method to generate and add stage name internally.
+
+        Args:
+            stage_name: Stage name to be added.
+        """
+
+        if stage_name is None:
+            stage_name = self.curr_stage_name if self.curr_stage_name is not None else "stage %s" % self.nstages
+
+        if not (stage_name in self.input_settings.keys()):
+            self.input_settings[stage_name] = OrderedDict()
+            self.curr_stage_name = stage_name  # type: ignore
+
+    def _add_command(self, command: str, args: str = None, stage_name: str = None):
+        """
+        Helper method to add a single LAMMPS command and its arguments to
+        LAMMPS input file.
+
+        Args:
+            command: LAMMPS command
+            args: Arguments for the LAMMPS command
+            stage_name: If a stage name is mentioned, the command is added
+                under that stage block else latest stage is assumed.
+        """
+        if self.curr_stage_name is None:
+            self.init_stage()
+        self._add_stage_name(stage_name)
+
+        if args is None:
+            string_split = command.split()
+            command = string_split[0]
+            args = " ".join(string_split[1:])
+        self.input_settings[self.curr_stage_name][command] = args
```

Comment:
I think there might be potential for unintended behavior here. What if I pass a `stage_name` kwarg that is different than `self.curr_stage` (which already has a value, not `None`) AND `stage_name` is already in `self.input_settings`?

If I follow the logic correctly, `add_stage_name` would not do anything in that case, and the command would be added under the key `self.curr_stage`  and not `stage_name`.